pkg/steps/git_source.go

```diff
@@ -0,0 +1,65 @@
+package steps
+
+import (
+	"context"
+	"fmt"
+
+	buildapi "github.com/openshift/api/build/v1"
+	imageclientset "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
+
+	"github.com/openshift/ci-operator/pkg/api"
+)
+
+type gitSourceStep struct {
+	config      api.ProjectDirectoryImageBuildInputs
+	resources   api.ResourceConfiguration
+	imageClient imageclientset.ImageV1Interface
+	buildClient BuildClient
+	jobSpec     *api.JobSpec
+}
+
+func (s *gitSourceStep) Inputs(ctx context.Context, dry bool) (api.InputDefinition, error) {
+	return s.jobSpec.Inputs(), nil
+}
+
+func (s *gitSourceStep) Run(ctx context.Context, dry bool) error {
+	return handleBuild(s.buildClient, buildFromSource(s.jobSpec, "", api.PipelineImageStreamTagReferenceRoot, buildapi.BuildSource{
+		Type:       buildapi.BuildSourceGit,
+		ContextDir: s.config.ContextDir,
+		Git: &buildapi.GitBuildSource{
+			URI: fmt.Sprintf("https://github.com/%s/%s.git", s.jobSpec.Refs.Org, s.jobSpec.Refs.Repo),
+			Ref: s.jobSpec.Refs.BaseRef,
```

Comment:
Technically this should be the merge commit of all the pulls and the base ref, but we have no way to do that with the OpenShift `Build`. We also can't just use the PR commit as that won't work on batches. We should add to the doc that you will get the build from the latest of the target branch, which is the same behavior as with a separate config. So technically still annoying but not a regression. Still possible to change the image build in the repo in a way where it won't break tests on the PR but will cause all subsequent tests to fail.